packages/cli/src/commands/domains/index.ts

```diff
@@ -70,18 +74,68 @@ www.example.com  CNAME            www.example.herokudns.com
     return tableConfig
   }
 
+  getFilteredDomains = (filterKeyValue: string, domains: Array<Heroku.Domain>) => {
```

Comment:
Since the filtering from the `--filter` flag happens inside of `ux.table()`, I had to add filtering functionality here so that the `... over 100 domain` warning and prompt accurately showed the amount of domains available.